include/llbuild/BuildSystem/BuildSystem.h

```diff
@@ -180,6 +180,14 @@ class BuildSystemDelegate {
   /// \param data - The error message.
   virtual void commandHadError(Command*, StringRef data) = 0;
 
+  /// Called to report an error that relates to multiple commands.
+  ///
+  /// For example, an error where multiple commands are trying to
+  /// build the same node.
+  ///
+  /// \param data - The error message.
+  virtual void commandsHadError(std::vector<Command*>, StringRef data) = 0;
```

Comment:
I think we want to move away from generic strongly-typed error messages, and instead move to reporting different types of diagnostics directly to the client with all the information they need to present a good error.

Given that direction, I think we should probably avoid introducing a `commandsHadError` type hook.